libs/python/interpreters.go

```diff
@@ -0,0 +1,191 @@
+			return nil, fmt.Errorf("listing %s: %w", prefix, err)
+		}
+		for _, v := range entries {
+			if v.IsDir() {
+				continue
+			}
+			if strings.Contains(v.Name(), "-") {
+				// skip python3-config, python3.10-config, etc
+				continue
+			}
+
+			// If Python3 is installed on Windows through GUI installer app that was
+			// downloaded from https://python.org/downloads/windows, it may appear
+			// in $PATH as `python`, even though it means Python 2.7 in all other
+			// operating systems (macOS, Linux).
+			//
+			// See https://github.com/databrickslabs/ucx/issues/281
+			if !strings.HasPrefix(v.Name(), "python") {
+				continue
+			}
+			bin := filepath.Join(prefix, v.Name())
+			resolved, err := filepath.EvalSymlinks(bin)
+			if err != nil {
+				log.Debugf(ctx, "cannot resolve symlink for %s: %s", bin, resolved)
+				continue
+			}
+			if seen[resolved] {
+				continue
+			}
+			seen[resolved] = true
+
+			// probe the binary version by executing it, like `python --version`
+			// and parsing the output.
+			//
+			// Keep in mind, that mswin installations get python.exe and pythonw.exe,
+			// which are slightly different: see https://stackoverflow.com/a/30313091
+			out, err := process.Background(ctx, []string{resolved, "--version"})
+			var processErr *process.ProcessError
+			if errors.As(err, &processErr) {
+				log.Debugf(ctx, "failed to check version for %s: %s", resolved, processErr.Err)
+				continue
+			}
+			if err != nil {
+				log.Debugf(ctx, "failed to check version for %s: %s", resolved, err)
+				continue
+			}
+
+			words := strings.Split(strings.TrimSpace(out), " ")
+			// The Python distribution from the Windows Store is available in $PATH as `python.exe`
+			// and `python3.exe`, even though it symlinks to a real file packaged with some versions of Windows:
+			// /c/Program Files/WindowsApps/Microsoft.DesktopAppInstaller_.../AppInstallerPythonRedirector.exe.
+			// Executing the `python` command from this distribution opens the Windows Store, allowing users to
+			// download and install Python. Once installed, it replaces the `python.exe` and `python3.exe`` stub
+			// with the genuine Python executable. Additionally, once user installs from the main installer at
+			// https://python.org/downloads/windows, it does not replace this stub.
+			//
+			// However, a drawback is that if this initial stub is run with any command line arguments, it quietly
+			// fails to execute. According to https://github.com/databrickslabs/ucx/issues/281, it can be
+			// detected by seeing just the "Python" output without any version info from the `python --version`
+			// command execution.
+			//
+			// See https://github.com/pypa/packaging-problems/issues/379
+			// See https://bugs.python.org/issue41327
+			if len(words) < 2 {
+				continue
+			}
+			if words[0] != "Python" {
+				continue
+			}
+			lastWord := words[len(words)-1]
+			version := semver.Canonical("v" + lastWord)
+			if version == "" {
+				continue
```

Comment:
Please log.Debugf the version that we did get for sanity. Or log it at trace level before the words check. Then we get to see the output of every invocation if we really need to.